web/src/search/input/QueryBuilder.test.tsx

```diff
@@ -0,0 +1,184 @@
+import * as React from 'react'
+import { cleanup, fireEvent, getBySelectText, getByTestId, queryByTestId, render, wait } from 'react-testing-library'
+import sinon from 'sinon'
+import { QueryBuilder } from './QueryBuilder'
+
+describe('QueryBuilder', () => {
+    afterAll(cleanup)
+
+    it('updating the "Repository" field fires the onFieldsQueryChange prop handler with the `repo:` filter', () => {
```

Comment:
```suggestion
    it('fires the onFieldsQueryChange prop handler with the `repo:` filter when updating the "Repository" field ', () => {
```

Make tests read like an English sentence starting with `it`